### PR TITLE
feat: Modify config keys from shell

### LINKED ISSUE #1080: Make it possible to modify the config from shell
#### Is your feature request related to a problem? Please describe.

Currently the configuration is static and cannot change unless you edit the file

#### Describe the solution you'd like

Commands like `starship configure set character.symbol "$"`, like [bspwm](https://github.com/baskerville/bspwm) is doing with their config

### PR DESCRIPTION
**Pull request by jletey**
#### Description

- Alias `starship configure` to `starship config`
- You can now pass in a key (e.g. `character.symbol`) and value (e.g. `"$"`) to `starship config` and it will update Starship's configuration file for you.

#### Motivation and Context

Closes #1080

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### How Has This Been Tested?

- [x] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [ ] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

**Comment by sandorex** [comment]
You should probably keep `starship configure` as an alias to `starship config`

**Comment by jletey** [comment]
@sandorex Good idea :+1:

File: src/configure.rs [hunk 1]
```diff
@@ -3,8 +3,55 @@ use std::ffi::OsString;
...
+
+fn get_file_path() -> Option<PathBuf> {
+    if let Ok(path) = env::var("STARSHIP_CONFIG") {
+        // Use $STARSHIP_CONFIG as the config path if available
+        Some(PathBuf::from(path))
+    } else {
+        // Default to using ~/.config/starship.toml
+        home_dir().map(|path| path.join(".config/starship.toml"))
+    }
+}
```
Comment (andytom) [inline src/configure.rs:53]: Would it be better to use the `get_config_path` function rather than duplicate the code here?

**Comment by jletey** [comment]
@andytom Thanks for taking a look at the PR 😊 You can now pass in integers and booleans and they will be parsed correctly. Also when you pass in an invalid configuration key, the program halts and "soft" errors.

**Comment by jletey** [comment]
@andytom I have made your suggested change. I think using `process::exit` is better than `process::abort`. It seems like the dotnet module tests are failing, but I guess this has nothing to do with my changes.

**Pull request merged**